Create tutorial: Running Commands in an Opam Switch (#1825)
ocaml/ocaml.org: data/tutorials/platform/bp_09_opam_path.md

---
id: "opam-path"
title: "Running Commands in an Opam Switch"
description: |

>   How to use commands provided by Apam in a switch

Reviewer: ```suggestion
  How to use commands installed in an Opam switch
```

2. Setup `direnv` integration

Add the following line to your shell profile (e.g., `~/.bashrc` or `~/.zshrc`):
```bash
$ eval $(direnv hook <shell>)
```
Replace `<shell>` with your actual shell type (`bash`, `zsh`, `fish`, etc.).

3. Configure opam with `direnv`

In your OCaml project directory, create a file named `.envrc` and add the following line to automatically load the opam environment:
```bash
eval $(opam env)
```

4. Allow `direnv`

Navigate to your project directory and run the following command to allow `direnv` to load the environment:
```bash
$ direnv allow
```

This command activates `direnv` for the current directory, ensuring that the opam switch environment is loaded whenever you enter the directory.

5. Usage

Now, whenever you navigate to your OCaml project directory, `direnv` will automatically activate the opam switch environment specified in your `.envrc` file. This eliminates the need to manually run `opam env` each time you work on your project.

6. Example

Suppose you have an OCaml project in folder `disco` and local opam switch is associated to it, and a `.envrc` file in that folder containing the following:
```bash
eval $(opam env)
```

> After running `direnv allow`, `direnv` will handle the opam switch activation for you whenever you.

Reviewer 2: This appears to be an incomplete sentence.

> After running `direnv allow`, `direnv` will handle the opam switch activation for you whenever you.

Reviewer: ```suggestion
After running `direnv allow`, `direnv` will handle the opam switch activation for you.
```
